docs/developer-docs/latest/development/custom-fields.md

```diff
@@ -0,0 +1,339 @@
+
+::: details Example: Registering an example "color" custom field in the admin panel:
+
+```jsx
+// path: ./src/plugins/color-picker/strapi-admin.js
+
+register(app) {
+  app.customFields.register({
+    name: "color",
+    pluginId: "color-picker", // the custom field is created by a color-picker plugin
+    type: "string", // the color will be stored as a string
+    intlLabel: {
+      id: "color-picker.color.label",
+      defaultMessage: "Color",
+    },
+    intlDescription: {
+      id: "color-picker.color.description",
+      defaultMessage: "Select any color",
+    } 
+    icon: ColorIcon,
+    components: {
+      Input: async () => import(/* webpackChunkName: "input-component" */ "./Input"),
+    } 
+    options: {
+      base: [
+        /*
+          Declare settings to be added to the "Base settings" section
+          of the field in the Content-Type Builder
+        */ 
+        {
+          sectionTitle: { // Add a "Format" settings section
+            id: 'color-picker.color.section.format',
+            defaultMessage: 'Format',
+          },
+          items: [ // Add settings items to the section
+            {
+              /*
+                Add a "Color format" dropdown
+                to choose between 2 different format options
+                for the color value: hexadecimal or RGBA
+              */
+              intlLabel: {
+                id: 'color-picker.color.format.label',
+                defaultMessage: 'Color format',
+              },
+              name: 'options.format',
+              type: 'select',
+              value: 'hex', // option selected by default
+              options: [ // List all available "Color format" options
+                {
+                  key: 'hex',
+                  value: 'hex',
+                  metadatas: {
+                    intlLabel: {
+                      id: 'color-picker.color.format.hex',
+                      defaultMessage: 'Hexadecimal',
+                    },
+                  },
+                },
+                {
+                  key: 'rgba',
+                  value: 'rgba',
+                  metadatas: {
+                    intlLabel: {
+                      id: 'color-picker.color.format.rgba',
+                      defaultMessage: 'RGBA',
+                    },
+                  },
+                },
+              ],
+            },
+          ],
+        },
+      ],
+      advanced: [
+        /*
+          Declare settings to be added to the "Advanced settings" section
+          of the field in the Content-Type Builder
+        */ 
+      ],
+      validator: args => ({
+        format: yup.string().required({
+          id: 'options.color-picker.format.error',
+          defaultMessage: 'The color format is required',
+        }),
+      })
+      }),
+    },
+  });
+}
+```
+
+:::
+
+::: note
+Relations, media, components, or dynamic zones can't be used as a custom field's `type` parameter.
+:::
```

Comment:
This seems quite important yet it's a bit hidden below the whole section. Could we move that inside the table so that users have all the information they need at the same place?
I know VuePress is annoying with callouts but maybe in the meantime we can just add a line break and then the note in the form of the emoji + the sentence? (I know I used this method in the [User Guide](https://docs.strapi.io/user-docs/latest/content-manager/writing-content.html#filling-up-fields) sometimes)

Reply:
Great idea, thanks! I initially wanted to include it in the table and didn't find a good solution. Your suggestion is 👌 , thank you!

Reply:
[Fixed](https://github.com/strapi/documentation/pull/1035/commits/639f19f20ce3c82851eb354086b5aee99a2f2c3f) ✅